components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +67,527 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
 }
 
 - (void)commonBannerViewInit {
-  self.backgroundColor = [UIColor whiteColor];
+  self.backgroundColor = MDCColorFromRGB(MDCBannerDefaultBackgroundColor);
+  _containerView = [[UIView alloc] initWithFrame:CGRectZero];
+  _containerView.translatesAutoresizingMaskIntoConstraints = NO;
+  [self addSubview:_containerView];
 
-  _buttons = [[NSArray alloc] init];
-  self.numberOfButtons = 1;
+  _buttons = [[NSMutableArray alloc] init];
 
   UILabel *textLabel = [[UILabel alloc] init];
-  textLabel.font = [UIFont systemFontOfSize:kTextFontSize];
-  textLabel.textColor = [UIColor colorWithWhite:0 alpha:kTextColorOpacity];
+  textLabel.translatesAutoresizingMaskIntoConstraints = NO;
+  textLabel.font = [MDCTypography body2Font];
+  textLabel.textColor = UIColor.blackColor;
+  textLabel.alpha = [MDCTypography body2FontOpacity];
   textLabel.numberOfLines = kTextNumberOfLineLimit;
   textLabel.lineBreakMode = NSLineBreakByTruncatingTail;
   _textLabel = textLabel;
-  [self addSubview:textLabel];
+
+  _isLayoutDirty = YES;
+
+  [self.containerView addSubview:textLabel];
 }
 
 #pragma mark - Property Setter and Getter
 
 - (void)setText:(NSString *)text {
   self.textLabel.text = text;
+  self.isLayoutDirty = YES;
 }
 
 - (NSString *)text {
   return self.textLabel.text;
 }
 
-- (void)setImage:(UIImage *)image {
-  if (image) {
-    _image = image;
-    if (!self.iconImageViewContainer) {
-      CGRect iconImageContainerFrame =
-          CGRectMake(0, 0, kIconImageContainerSideLength, kIconImageContainerSideLength);
-      UIView *iconImageViewContainer = [[UIView alloc] initWithFrame:iconImageContainerFrame];
-      UIImageView *iconImageView = [[UIImageView alloc] initWithImage:image];
-      CGRect iconImageFrame = CGRectMake(0, 0, kIconImageSideLength, kIconImageSideLength);
-      iconImageView.frame = iconImageFrame;
-      iconImageView.contentMode = UIViewContentModeScaleAspectFit;
-      iconImageView.clipsToBounds = YES;
-      [iconImageViewContainer addSubview:iconImageView];
-      self.iconImageViewContainer = iconImageViewContainer;
-      self.iconImageView = iconImageView;
-      [self addSubview:iconImageViewContainer];
-    } else {
-      self.iconImageView.image = image;
-    }
+- (void)setIcon:(UIImage *)icon {
+  if (icon == nil) {
+    [self.iconImageView removeFromSuperview];
+    self.iconImageView = nil;
+    self.isLayoutDirty = YES;
```

Comment:
This can be removed and instead we can have `[self invalidateIntrinsicContentSize]`

Reply:
Same reason as mentioned above in https://github.com/material-components/material-components-ios/pull/6830#discussion_r264361551. Marking as resolved to keep the conversation in a single thread.